src/polyswarmd/eth.py

```diff
@@ -169,11 +170,20 @@ def build_transaction(call, nonce):
     options = {
         'nonce': nonce,
         'chainId': int(g.chain.chain_id),
-        'gas': gas_limit,
+        'gas': MAX_GAS_LIMIT,
     }
+
     if g.chain.free:
         options["gasPrice"] = 0
 
+    try:
+        gas = int(call.estimateGas({'from': g.eth_address, **options}))
+        options['gas'] = min(MAX_GAS_LIMIT, gas)
```

Comment:
Kinda worried about nonce desyncing issues if the transaction gas is less than `MAX_GAS_LIMIT` and etimateGas produces a gas that's higher the previous with the same nonce. Would that be an issue for us? Might need to see this in action to really know. 

I do know the `etimateGas` can produce somewhat wonky results based on the contract state.
https://ethereum.stackexchange.com/questions/266/what-are-the-limitations-to-estimategas-and-when-would-its-estimate-be-considera

Reply:
gas limit doesnt affect replacement txs only gas price afiak

usually the value will be close unless it fails in which case we fall back to max gas limit, we're supposed to addd 20% or so to be safe but left out that mult

Reply:
Okay sounds good